후/src/mission2_0/RandomGacha.java

```diff
@@ -0,0 +1,82 @@
+package mission2_0;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+class RandomGacha {
+
+    final static int MIN_COIN = 1;
+
+    Scanner coinSlot;
+    private int coin = 0;
+    List<CocoaMember> members;
+
+    RandomGacha() {
+        coinSlot = new Scanner(System.in);
+        this.members = CocoaMember.makeShuffledSquad();
+    }
+
+    boolean isValidInput(String input) {
+        try {
+            if (Integer.parseInt(input) < MIN_COIN) {
+                throw new InputMismatchException();
+            }
+            return true;
+        } catch (NumberFormatException e) {
+            System.out.println("숫자로 입력");
+            return false;
+        } catch (InputMismatchException e) {
+            System.out.println("0보다 큰 숫자로 입력");
+            return false;
+        }
+    }
+
+    void getUserCoin() {
+        String input = "";
+
+        System.out.print("insert coin >>>>> ");
+        input = coinSlot.nextLine();
+
+        if (isValidInput(input)) {
+            int coins = Integer.parseInt(input);
+            insertCoin(coins);
+            coinSlot.close();
+            return;
+        }
+
+        getUserCoin();
+    }
+
+    void insertCoin(int coins) {
+        this.coin += coins;
+    }
+
+    void pick() {
+        StringBuilder result = new StringBuilder();
+        Stream<CocoaMember> memberStream;
+        memberStream = members.stream();
+
+        memberStream
+            .limit(coin)
+            .peek(member -> coin--)
+            .forEach(member -> result.append(member.getName() + ", "));
+
+        System.out.println(result.substring(0, result.length() - 2));
```

Comment:
코코아 단계는 넘어가는 것 같지만, 기왕 stream 을 사용하셨다면 이 모두를 한번에 처리할 수 있을 것 입니다.

Reply:
> 코코아 단계는 넘어가는 것 같지만, 기왕 stream 을 사용하셨다면 이 모두를 한번에 처리할 수 있을 것 입니다.

스트림을 사용하지 않는 쪽으로 고치긴 했는데.. 스트림을 사용한다면  어떻게 한번에 처리할 수 있을지 고민을 해보아도 잘 모르겠습니다ㅠㅠ

Reply:
```suggestion
  members.stream()
	  .limit(coin)
	  .forEach(e -> coin--);
  
  final String result = members.stream()
	  .limit(coin)
	  .map(CocoaMember::getName)
	  .collect(Collectors.joining(", "));
  System.out.println(result);
```

돌려보진 않긴 했는데, 이런 느낌으로 하시면 될 것 같습니다.
그리고 전체 로직을 파악하지는 못했지만 `coin--` 하는 건 뭔가 좀 불편하네요..